### PR TITLE
chore(medusa): Add seed script for isolated product + fixes

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
- Create a new script that allow us to populate the product especially when we are working in isolation
- fix addProducts to shipping profile when isolation flag is enabled
- When attaching inventory items to variant, there shouldn't be a constraint check in isolation (for now?)
- fix product module unique constraints on deleted entities
- fix create cart workflow sales channels management, guarded by flag
- add scripts to assign product to profile
- fix cart update line item confirm inventory
- fix shipping retrieval from cart
- fix shipping profile service get map and the impact
- fix cart completion
- fix reservation variant retrieval
- fix reservation when variant does not manager inventory and there is no inventory module

**Flow to follow to tests the checkout flow in an isolated mode**

- set your flags such as workflows is true, sales channels is false, isolated product domain is true
```js
featureFlags: {
    sales_channels: false,
    isolate_product_domain: true,
    workflows: true,
  },
```
- Config the product module
```js
modules: {
    // ...
    productService: {
      scope: "internal",
      resources: "isolated",
      resolve: "@medusajs/product",
      options: {
        database: {
          clientUrl: "postgresql://localhost:5432/medusa-store"
        }
      }
    },
  }
```
- update the env file 
```bash
PRODUCT_POSTGRES_URL=<your url>
```
- run the migrations
```bash
medusa migrations run
```
- run the product module migration
```bash
node node_modules/@medusajs/product/dist/scripts/bin/run-migration-up.js
```
- run the seed
```bash
yarn seed
```
- run the product isolated seed
```bash
node node_modules/@medusajs/medusa/dist/scripts/seed-isolated-products.js
```
- run the product shipping profile isolated seed
```bash
node node_modules/@medusajs/medusa/dist/scripts/add-product-to-shipping-profiles-isolated.js
```

**NOTE**
The only flow that does not work is without the inventory module but with manage_inventory.

### CODE REVIEW

File: .changeset/selfish-sheep-exist.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@medusajs/medusa": patch
+---
+
+chore(medusa): Add seed script for isolated product + fixes
```
Comment (carlos-r-l-rodrigues) [inline .changeset/selfish-sheep-exist.md:7]: add patch to `/products` and `/workflows`

File: packages/medusa/src/scripts/add-product-to-shipping-profiles-isolated.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      products = await productModuleService.list(
+        {},
+        {
+          select: ["id"],
+          skip,
+          take,
+        }
+      )
+
+      skip += products.length
```
Comment (carlos-r-l-rodrigues) [inline packages/medusa/src/scripts/add-product-to-shipping-profiles-isolated.ts]: this should be before line 72

File: packages/medusa/src/scripts/add-product-to-shipping-profiles-isolated.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    if (!productModuleService) {
+      throw new Error(
+        `Cannot run ${Workflows.CreateProducts} workflow without '@medusajs/product' installed`
+      )
+    }
+
+    let skip = 0
+    let take = 100
```
Comment (carlos-r-l-rodrigues) [inline packages/medusa/src/scripts/add-product-to-shipping-profiles-isolated.ts]: it's safe to bump to 500

**Review by carlos-r-l-rodrigues — APPROVED** [review]
LGTM

**Review by fPolic — APPROVED** [review]
LGTM!

File: packages/medusa/src/services/shipping-profile.ts [hunk 4]
```diff
@@ -526,7 +514,11 @@ class ShippingProfileService extends TransactionBaseService {
       const productShippinProfileMap = await this.getMapProfileIdsByProductIds(
         cart.items.map((item) => item.variant?.product_id)
       )
-      profileIds = new Set([...productShippinProfileMap.values()])
+      profileIds = new Set(
+        Object.values(productShippinProfileMap)
```
Comment (fPolic) [inline packages/medusa/src/services/shipping-profile.ts]: todo: typo (in a couple of places)

**Comment by adrien2p** [comment]
this pr is closed as everything has been merged into https://github.com/medusajs/medusa/pull/5148

**Pull request closed without merging**